contracts/v2/Tipping_v2.aes

```diff
@@ -89,7 +90,8 @@ contract Tipping =
       url_index = {},
       tips = {},
       retips = {},
-      oracle_service = oracle_service }
+      oracle_service = oracle_service,
+      version = "v2" }
```

Comment:
Isn't it better to pass the version number as argument to the `init` call? 🤔

Reply:
no, its good to be hardcoded, as we would need a new file for any future contract version, as we continue support any old version.

Reply:
v1 and v2 handling will also be hardcoded in certain places, same for future contracts that eventually support newer features

Reply:
Is there any interface check done rather than just this version number? 

What would happen if I deploy a contract in the future which introduces new functionality or breaks backwards compatibility and I do not bump the version number? 🤔

Reply:
> What would happen if I deploy a contract in the future which introduces new functionality or breaks backwards compatibility and I do not bump the version number?

It is our setup, the version number is just to make it easy for us, we can make an interface check or for every fetching state pass a mapping of version to contract address.

If we forget it we made a mistake and can deploy a new contract.

But in any case if there is a new version of the contract it's a new file with new tests and "vX" will have to be a testcase there.